multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/PollExecuteAppStatusExecution.java

```diff
@@ -56,114 +57,124 @@ String getMessage() {
 
     private static final String DEFAULT_SUCCESS_MARKER = "STDOUT:SUCCESS";
     private static final String DEFAULT_FAILURE_MARKER = "STDERR:FAILURE";
-
-    private final RecentLogsRetriever recentLogsRetriever;
-
-    public PollExecuteAppStatusExecution(RecentLogsRetriever recentLogsRetriever) {
-        this.recentLogsRetriever = recentLogsRetriever;
-    }
+    private static final String LOGS_OFFSET_FOR_APP = "logsOffsetForAppExecution";
 
     @Override
     public AsyncExecutionState execute(ProcessContext context) {
         List<ApplicationStateAction> actions = context.getVariable(Variables.APP_STATE_ACTIONS_TO_EXECUTE);
         if (!actions.contains(ApplicationStateAction.EXECUTE)) {
             return AsyncExecutionState.FINISHED;
         }
-        CloudApplicationExtended app = getNextApp(context);
+        CloudApplicationExtended app = context.getVariable(Variables.APP_TO_PROCESS);
         CloudControllerClient client = context.getControllerClient();
         ApplicationAttributes appAttributes = ApplicationAttributes.fromApplication(app, app.getEnv());
-        AppExecutionDetailedStatus status = getAppExecutionStatus(context, client, appAttributes, app);
+
+        OffsetDateTime logsOffset = getLogOffsetAdapter(context);
+        List<ApplicationLog> recentLogs = client.getRecentLogs(app.getName(), logsOffset);
+        setLogsOffset(context, recentLogs);
+
+        AppExecutionDetailedStatus status = getAppExecutionStatus(context, appAttributes, recentLogs);
         ProcessLoggerProvider processLoggerProvider = context.getStepLogger()
                                                              .getProcessLoggerProvider();
-        StepsUtil.saveAppLogs(context, client, recentLogsRetriever, app.getName(), LOGGER, processLoggerProvider);
+        StepsUtil.saveAppLogs(context, client, app.getName(), LOGGER, processLoggerProvider);
```

Comment:
Are the logs saved Ascending order? We're getting them in descending

Reply:
Yes, they are sorted and saved correctly